exportReport/ExtensionExportReport.java

```diff
@@ -0,0 +1,298 @@
+ * MOD DESC		:	
+ */
+
+public class ExtensionExportReport extends ExtensionAdaptor
+{
+
+	public static final String			NAME				= "ExtensionExportReport";
+	private ZapMenuItem					menuExportReport	= null;
+	private FrameExportReport			frameER				= null;
+
+	protected final Map<String, String>	alertTypeRisk		= new HashMap<>();
+
+	/* Nav Panel creation. */
+	private PanelSource					cardSource			= null;
+	private PanelAlertRisk				cardAlertRisk		= null;
+	private PanelAlertDetails			cardAlertDetails	= null;
+
+	/* Field Limit Constants */
+	private final int					textfieldLimit		= 30;
+	private final int					textareaLimit		= 600;
+
+	/* Relational CONSTANT for export types. */
+	private ArrayList<String>			alertSeverity		= new ArrayList<String>();
+	private ArrayList<String>			alertDetails		= new ArrayList<String>();
+	private ArrayList<String>			alertAdditional		= new ArrayList<String>();
+	private int							maxList				= 0;
+	
+	public ExtensionExportReport()
+	{
+		super();
+		initialize();
+	}
+
+	public ExtensionExportReport(String name)
+	{
+		super(name);
+	}
+
+	private void initialize()
+	{
+		this.setName("ExtensionExportReport");
+	}
+
+	@Override
+	public void hook(ExtensionHook extensionHook)
+	{
+		super.hook(extensionHook);
+		if (getView() != null)
+		{
+			extensionHook.getHookMenu().addReportMenuItem(getMenuExportReport());
+		}
+	}
+
+	@Override
+	public boolean canUnload()
+	{
+		return true;
+	}
+
+	@Override
+	public void unload()
+	{
+		super.unload();
+		if (frameER != null)
+		{
+			frameER.dispose();
+			frameER = null;
+		}
+	}
+
+	private ZapMenuItem getMenuExportReport()
+	{
+		if (menuExportReport == null)
+		{
+			menuExportReport = new ZapMenuItem("menu.report.html.generate");
+			menuExportReport.setText(Constant.messages.getString("exportReport.menu.export"));
+			menuExportReport.addActionListener(new java.awt.event.ActionListener()
+			{
+				@Override
+				public void actionPerformed(java.awt.event.ActionEvent e)
+				{
+					if (frameER != null)
+					{
+						frameER.requestFocusInWindow();
+						return;
+					}
+					getNewOptionFrame();
+					
+					// -Change the default close operation
+					WindowAdapter exitListener = new WindowAdapter() {
+					    @Override
+					    public void windowClosing(WindowEvent e) {
+					    	emitFrame();
+					    }
+					};
+					frameER.addWindowListener(exitListener);
+					frameER.setSize(400, 500);
+					frameER.setResizable(false);
+					frameER.setVisible(true);
+					frameER.centerFrame();//frameER.setLocationRelativeTo(null);
+				}
+			});
+		}
+		return menuExportReport;
+	}
+
+	public void getNewOptionFrame()
+	{
+		initializeLists();
+		frameER = new FrameExportReport(this, extensionGetCardSource(),extensionGetCardAlertRisk(),extensionGetCardAlertDetails());
+	}
+
+	private void initializeLists()
+	{
+		// PanelAlertRisk
+		alertSeverity.clear();
+		alertSeverity.add(Constant.messages.getString("exportReport.risk.severity.high"));
```

Comment:
Like in this case, because it's a rather large plugin i use Ctrl+F and search for `// PanelAlertRisk` to navigate quicker, if you'd like me to remove it i have no issue with that.